### PR TITLE
Add typer app command for generate

### PR DESCRIPTION
**Pull request by feluelle**
# Description

## What is the current behavior?

Currently, we don't have a CLI command for generating a DAG - only through the `generate_dag.py` script.

closes: #1008

## What is the new behavior?

We are adding a typer command for generating a DAG.

- refactor Makefile
- add missing deps
- add airflow and astro-sdk to optional deps; we install them via makefile and in dev/test/type_check nox sessions
- return dag file path in generate_dag
- use rich for printing
- improve tests
- remove unused project init and cli entrypoint
- add pytest-freezegun as dev deps
- set static start_date
- fix templates

## Does this introduce a breaking change?

No.

### Checklist
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: sql-cli/noxfile.py [hunk 1]
```diff
@@ -16,19 +16,13 @@ def dev(session: nox.Session) -> None:
     development environment to ``.nox/dev``.
     """
     session.install("nox")
-    session.install("-e", "../python-sdk/.[all]")
-    session.install("-e", "../python-sdk/.[tests]")
-    session.install("-e", ".[all]")
-    session.install("-e", ".[tests]")
+    session.install("-e", ".[all,tests]")
```
Comment (feluelle) [inline sql-cli/noxfile.py:19]: The `python-sdk` will be installed through `all`.

File: sql-cli/pyproject.toml [hunk 2]
```diff
@@ -34,6 +36,10 @@ tests = [
     "pytest-cov",
     "mypy"
 ]
+all = [
+    "apache-airflow>=2.0",
+    "astro-sdk-python[all]>=1.0.0",
+]
```
Comment (feluelle) [inline sql-cli/pyproject.toml:42]: I have added these two as optional dependencies. For example, they are already useful to validate the syntax of the generated DAG.

I would suggest we make them required dependencies when we have `flow run`.

File: sql-cli/sql_cli/__main__.py [hunk 3]
```diff
@@ -1,24 +1,41 @@
...
+app = Typer(add_completion=False)
 
 
 @app.command()
 def version() -> None:
     """
     Print the SQL CLI version.
     """
-    typer.echo(f"Astro SQL CLI {sql_cli.__version__}")
+    rprint("Astro SQL CLI", __version__)
```
Comment (feluelle) [inline sql-cli/sql_cli/__main__.py:17]: using rich is [recommended](https://typer.tiangolo.com/tutorial/printing/).

File: sql-cli/noxfile.py (hunk 1 above)
Comment (tatiana) [reply sql-cli/noxfile.py:19]: The advantage of the previous way is that we'd be testing the latest version of the Python SDK, as opposed to the last released version. Perhaps we should have both, WDYT?

Comment (feluelle) [reply sql-cli/noxfile.py:19]: Not sure if this is an advantage tbh. We should, in my opinion, treat `python-sdk` as any other package. We should only test against released stable versions.

Comment (feluelle) [reply sql-cli/noxfile.py:19]: In my opinion testing against non-stable versions does not make sense as they could quickly change.

Comment (tatiana) [reply sql-cli/noxfile.py:19]: So if the `python-sdk`  introduces a breaking change, the SQL CLI would only know after it is released?

Comment (feluelle) [reply sql-cli/noxfile.py:19]: If we use proper semver in python-sdk, this is not a problem in my opinion. We could restrict the version also the same we do for other dependencies.

**Review by tatiana — APPROVED** [review]

File: sql-cli/Makefile [hunk 4]
```diff
@@ -32,21 +31,14 @@ virtualenv:  ## Create Python virtualenv
...
 install: virtualenv  ## Install python dependencies in existing virtualenv
 	@echo "Installing Python dependencies using $(PIP)"
 	@$(PIP) install --upgrade pip
 	@$(PIP) install nox
 	@$(PIP) install pre-commit
-	@$(PIP) install -e .[all]
-	@$(PIP) install .[tests]
+	@$(PIP) install -e .[all,tests]
 
-setup: system install  ## Setup a local development environment
```
Comment (dimberman) [inline sql-cli/Makefile:49]: Is this relevent to the current PR or should it be a different PR?

File: sql-cli/examples/_dags/advanced.py [hunk 5]
```diff
@@ -2,11 +2,11 @@
 from airflow import DAG
 from airflow.utils import timezone
 from astro import sql as aql
-from astro.table import Table
+from astro.sql.table import Table
 
 with DAG(
     dag_id="advanced",
-    start_date=timezone.parse("2022-10-05 06:47:15.790544"),
+    start_date=timezone.parse("2022-10-10 09:11:57.308687"),
```
Comment (dimberman) [inline sql-cli/examples/_dags/advanced.py:9]: Why this change?

File: sql-cli/examples/_dags/simple.py [hunk 6]
```diff
@@ -2,11 +2,11 @@
 from airflow import DAG
 from airflow.utils import timezone
 from astro import sql as aql
-from astro.table import Table
+from astro.sql.table import Table
```
Comment (dimberman) [inline sql-cli/examples/_dags/simple.py:5]: Why this change?

File: sql-cli/noxfile.py (hunk 1 above)
Comment (dimberman) [reply sql-cli/noxfile.py:19]: I'm a -1 on this. There will be situations where we are both changing the SQL SDK and the python SDK and we will need to test the changes together.

**Review by dimberman — CHANGES_REQUESTED** [review]

File: sql-cli/noxfile.py (hunk 1 above)
Comment (feluelle) [reply sql-cli/noxfile.py:19]: I agree that this may need its own PR.

File: sql-cli/examples/_dags/advanced.py (hunk 5 above)
Comment (feluelle) [reply sql-cli/examples/_dags/advanced.py:9]: This is the time of the last auto-generated example. I ran it because of the change you see below. Let's continue the discussion in https://github.com/astronomer/astro-sdk/pull/1032#discussion_r991511811.

File: sql-cli/examples/_dags/simple.py (hunk 6 above)
Comment (feluelle) [reply sql-cli/examples/_dags/simple.py:5]: Because, in my opinion we should use the released python-sdk version - not the current one. Otherwise it will become very confusing for users if they see the example dag of an unreleased version. The import path change was introduced in #1013
